library/src/main/java/com/novoda/downloadmanager/LiteDownloadFileId.java

```diff
@@ -2,41 +2,40 @@
 
 final class LiteDownloadFileId implements DownloadFileId {
 
-    private final int id;
+    private final String id;
 
-    LiteDownloadFileId(int id) {
+    LiteDownloadFileId(String id) {
         this.id = id;
     }
 
+    @Override
+    public String rawId() {
+        return String.valueOf(id);
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (!(o instanceof LiteDownloadFileId)) {
+        if (o == null || getClass() != o.getClass()) {
             return false;
         }
 
         LiteDownloadFileId that = (LiteDownloadFileId) o;
 
-        return id == that.id;
-
-    }
-
-    @Override
-    public String toRawId() {
-        return String.valueOf(id);
+        return id != null ? id.equals(that.id) : that.id == null;
     }
 
     @Override
     public int hashCode() {
-        return id;
+        return id != null ? id.hashCode() : 0;
```

Comment:
This comes from outside of the library so it is completely feasible that this could be null.